commit f098addbdb44c8a565367f5162f3ab170ed9404a
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date:   Thu Jul 1 13:39:15 2021 -0300

    tools headers cpufeatures: Sync with the kernel sources
    
    To pick the changes from:
    
      f43b9876e857c739 ("x86/retbleed: Add fine grained Kconfig knobs")
      a149180fbcf336e9 ("x86: Add magic AMD return-thunk")
      15e67227c49a5783 ("x86: Undo return-thunk damage")
      369ae6ffc41a3c11 ("x86/retpoline: Cleanup some #ifdefery")
      4ad3278df6fe2b08 x86/speculation: Disable RRSBA behavior
      26aae8ccbc197223 x86/cpu/amd: Enumerate BTC_NO
      9756bba28470722d x86/speculation: Fill RSB on vmexit for IBRS
      3ebc170068885b6f x86/bugs: Add retbleed=ibpb
      2dbb887e875b1de3 x86/entry: Add kernel IBRS implementation
      6b80b59b35557065 x86/bugs: Report AMD retbleed vulnerability
      a149180fbcf336e9 x86: Add magic AMD return-thunk
      15e67227c49a5783 x86: Undo return-thunk damage
      a883d624aed463c8 x86/cpufeatures: Move RETPOLINE flags to word 11
      51802186158c74a0 x86/speculation/mmio: Enumerate Processor MMIO Stale Data bug
    
    This only causes these perf files to be rebuilt:
    
      CC       /tmp/build/perf/bench/mem-memcpy-x86-64-asm.o
      CC       /tmp/build/perf/bench/mem-memset-x86-64-asm.o
    
    And addresses this perf build warning:
    
      Warning: Kernel ABI header at 'tools/arch/x86/include/asm/cpufeatures.h' differs from latest version at 'arch/x86/include/asm/cpufeatures.h'
      diff -u tools/arch/x86/include/asm/cpufeatures.h arch/x86/include/asm/cpufeatures.h
      Warning: Kernel ABI header at 'tools/arch/x86/include/asm/disabled-features.h' differs from latest version at 'arch/x86/include/asm/disabled-features.h'
      diff -u tools/arch/x86/include/asm/disabled-features.h arch/x86/include/asm/disabled-features.h
    
    Cc: Adrian Hunter <adrian.hunter@intel.com>
    Cc: Borislav Petkov <bp@suse.de>
    Cc: Ian Rogers <irogers@google.com>
    Cc: Jiri Olsa <jolsa@kernel.org>
    Cc: Namhyung Kim <namhyung@kernel.org>
    Cc: Peter Zijlstra <peterz@infradead.org
    Link: https://lore.kernel.org/lkml/YtQM40VmiLTkPND2@kernel.org
    Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>

diff --git a/tools/arch/x86/include/asm/disabled-features.h b/tools/arch/x86/include/asm/disabled-features.h
index 36369e76cc63..33d2cd04d254 100644
--- a/tools/arch/x86/include/asm/disabled-features.h
+++ b/tools/arch/x86/include/asm/disabled-features.h
@@ -50,6 +50,25 @@
 # define DISABLE_PTI		(1 << (X86_FEATURE_PTI & 31))
 #endif
 
+#ifdef CONFIG_RETPOLINE
+# define DISABLE_RETPOLINE	0
+#else
+# define DISABLE_RETPOLINE	((1 << (X86_FEATURE_RETPOLINE & 31)) | \
+				 (1 << (X86_FEATURE_RETPOLINE_LFENCE & 31)))
+#endif
+
+#ifdef CONFIG_RETHUNK
+# define DISABLE_RETHUNK	0
+#else
+# define DISABLE_RETHUNK	(1 << (X86_FEATURE_RETHUNK & 31))
+#endif
+
+#ifdef CONFIG_CPU_UNRET_ENTRY
+# define DISABLE_UNRET		0
+#else
+# define DISABLE_UNRET		(1 << (X86_FEATURE_UNRET & 31))
+#endif
+
 #ifdef CONFIG_INTEL_IOMMU_SVM
 # define DISABLE_ENQCMD		0
 #else
@@ -82,7 +101,7 @@
 #define DISABLED_MASK8	(DISABLE_TDX_GUEST)
 #define DISABLED_MASK9	(DISABLE_SGX)
 #define DISABLED_MASK10	0
-#define DISABLED_MASK11	0
+#define DISABLED_MASK11	(DISABLE_RETPOLINE|DISABLE_RETHUNK|DISABLE_UNRET)
 #define DISABLED_MASK12	0
 #define DISABLED_MASK13	0
 #define DISABLED_MASK14	0